src/lib/core/CHIPConfig.h

```diff
@@ -1926,6 +1926,28 @@
 #define CHIP_CONFIG_TEST 0
 #endif // CHIP_CONFIG_TEST
 
+/**
+ *  @def CHIP_CONFIG_ERROR_CLASS
+ *
+ *  If 0, #CHIP_ERROR is an integer type, ::chip::ChipError::StorageType.
+ *  If 1, #CHIP_ERROR is a class type, ::chip::ChipError.
+ */
+#ifndef CHIP_CONFIG_ERROR_CLASS
+#define CHIP_CONFIG_ERROR_CLASS 0
+#endif // CHIP_CONFIG_ERROR_CLASS
+
+/**
+ *  @def CHIP_CONFIG_ERROR_FORMAT_AS_STRING
+ *
+ *  If 0, then ChipError::FormatError() returns an integer (ChipError::BaseType).
+ *  If 1, then ChipError::FormatError() returns a const char *, from chip::ErrorStr().
+ *  In either case, the macro CHIP_ERROR_FORMAT expands to a suitable printf format.
+ */
+
+#ifndef CHIP_CONFIG_ERROR_FORMAT_AS_STRING
+#define CHIP_CONFIG_ERROR_FORMAT_AS_STRING 0
```

Comment:
Given the various places this patch replaces ErrorStr() bits with things that depend on this define, shouldn't this be set to 1 by default to preserve existing behavior?

Reply:
All uses of FormatError() in this PR replace integer formatting (unless by mistake), so 0 preserves existing behaviour. I haven't tried replacing existing uses of ErrorStr(), though I'm inclined to do so later for code that might run on small devices.

Reply:
> All uses of FormatError() in this PR replace integer formatting (unless by mistake)

I am looking specifically at the changes in `Device::OnSessionEstablished`, `ServerBase::Listen`, arguably `PASESession::HandleErrorMsg` (though the old code was clearly buggy).  I guess if those are changed back to explicit `ErrorStr` this can default to 0 with no behavior regressions.

Reply:
Ah yes, I got ahead of myself in a few places.